client/rest-high-level/src/test/java/org/elasticsearch/client/IndicesClientIT.java

```diff
@@ -1254,6 +1260,35 @@ public void testPutTemplate() throws Exception {
         assertThat((Map<String, String>) extractValue("my-template.aliases.alias-1", templates), hasEntry("index_routing", "abc"));
         assertThat((Map<String, String>) extractValue("my-template.aliases.{index}-write", templates), hasEntry("search_routing", "xyz"));
     }
+    
+    @SuppressWarnings("unchecked")
+    public void testPutUntypedTemplate() throws Exception {
```

Comment:
Also for consistency with 'put mapping', I think we should rename `testPutTemplate` -> `testPutTemplateWithTypes`, and then rename this one `testPutUntypedTemplate` -> `testPutTemplate`.